### PR TITLE
emmie

### PR DESCRIPTION
**Pull request by emmiehayes**
What do you think we were trying to get you to learn with this challenge?
  problem solving without a clearly defined spec

Outline in words (at least 5-6 sentences) what your code is doing in order to solve this problem.
   So my understanding wasn't clear with regards to all of the different 'cost' for the activity so I just 
 started with a single total_cost that I would pass in with the name.  Participants would pay to 
 participate in the activity based on what they were willing to pay and then based on the number of 
 participants that participated in the activity, each person would either be refunded or asked to pay a 
 remaining balance.  Which makes very little sense in the real world but I ran with it regardless. I did 
 this in order to avoid analysis paralysis.  My goal was at the end of the reunion each person pays an 
 equal price per each activity they participated in. Due to time constraints I didn't edge case test as 
 much as I could have.  My favorite part was breaking out the overpaid and underpaid participants in 
 the activity class.  

What was the most difficult portion of this for you?
   forging my own way when i didn't fully understand the spec- just because i didn't know if it would 
   bite me in the ass later on.

What came naturally to you? 
  TDD.  I would write a test and then create the method and binding pry into the method line by line to 
   build out what i need from the method

### CODE REVIEW

File: lib/activity.rb [hunk 1]
```diff
@@ -1,2 +1,48 @@
...
+    @total_cost   = total_cost.to_f
+    @participants = {}
+  end
+
+  def add_participant(person_name, person_paid)
+    @participants[person_name] = person_paid
+  end
+
+  def price_per_person
+    (@total_cost/@participants.length if @participants.length >= 1).round(2)
```
Comment (mikedao) [inline lib/activity.rb]: Check your parens here

Comment (mikedao) [reply lib/activity.rb]: What happens if theres no participants. Is there a test that covers this case?

File: lib/activity.rb [hunk 2]
```diff
@@ -1,2 +1,48 @@
...
+  def participants_who_overpaid
+    participants_overpaid = @participants.group_by do |participant|
+      participant[1] >= price_per_person
+    end
+    participants_overpaid[true]
+  end
+
+  def calculate_balance
+    participants_who_underpaid.map do |participant|
+      paid = participant.pop
```
Comment (mikedao) [inline lib/activity.rb]: Its more efficient to create a new collection rather than to pop and push
